### PR TITLE
Split Checkout form component into smaller files (II)

### PR DESCRIPTION
**Pull request by Aljullu**
This is a follow-up of #3062 and part of the fixes I will add for #2319.

### How to test the changes in this Pull Request:

Do some smoke testing of the Checkout block:
* create a new page and add the Checkout block, change some attributes in the sidebar and verify changes take place
* verify the Checkout block works in the frontend: try changing your address, picking a different shipping and payment method, etc.
* verify you can make a purchase

### CODE REVIEW

File: assets/js/blocks/cart-checkout/checkout/form/address-step.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+					addressFieldsConfig={ addressFieldsConfig }
+					billingFields={ billingFields }
+					defaultAddressFields={ defaultAddressFields }
+					requirePhoneField={ requirePhoneField }
+					setPhone={ setPhone }
+					setShippingAsBilling={ setShippingAsBilling }
+					setShippingFields={ setShippingFields }
+					shippingAsBilling={ shippingAsBilling }
+					shippingFields={ shippingFields }
+					showPhoneField={ showPhoneField }
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/checkout/form/address-step.js:73]: I feel like we're passing too many props to each component, maybe we can move the hook to inside `ShippingFieldsStep` and `BillingFieldsStep`, this is unrelated to this PR, just an idea.

**Review by senadir — APPROVED** [review]

File: assets/js/blocks/cart-checkout/checkout/form/address-step.js (hunk 1 above)
Comment (Aljullu) [reply assets/js/blocks/cart-checkout/checkout/form/address-step.js:73]: I found some weird behavior when calling `useCheckoutAddress()` from `ShippingFieldsStep` and `BillingFieldsStep` separately, that's why I moved it one level up so the hook is shared by both components. I will proceed merging as-is, we can investigate it further in the future.
